test/metabase/util/debugging_test.clj

```diff
@@ -0,0 +1,43 @@
+(ns metabase.util.debugging-test
+  (:require  [clojure.test :refer :all]
+             [metabase.config :as config]))
+
+(def dbg-prn-used? (atom false))
+
+(defn dbg-prn*
+  "Implementation for the `dbg-prn` macro."
+  [p x]
+  (p x)
+  x)
+
+(defmacro dbg-prn
+  "`prn`s the argument and also returns it, which is useful for debugging. There is also a CI check to ensure no usages
+  of it sneak into master. With two arguments, uses the first as a printing function instead of `prn`."
+  ([x]
+   `(dbg-prn prn ~x))
+  ([p x]
+   `(do
+      (when-not (= (ns-name *ns*) 'metabase.util.debugging-test)
+        (reset! metabase.util.debugging-test/dbg-prn-used? true))
```

Comment:
Since resetting the atom is a side-effect here you might want to do it outside of the macroexpansion, e.g.

```clj
(when-not ...)
`(dbg-prn* ~p ~x)
```

Otherwise, it's only resetting the atom if the `dbg-prn` form actually gets called. If you put it somewhere that doesn't end up getting called (e.g. in a conditional), the test won't fail